app/scripts/lib/transaction/eip5792.ts

```diff
@@ -109,16 +125,39 @@ export async function processSendCalls(
   const securityAlertId = generateSecurityAlertId();
   const validateSecurity = validateSecurityHook.bind(null, securityAlertId);
 
-  const { batchId: id } = await addTransactionBatch({
-    from,
-    networkClientId,
-    origin,
-    securityAlertId,
-    transactions,
-    validateSecurity,
-  });
+  let batchId: Hex;
+  if (Object.keys(transactions).length === 1) {
+    const trxnParams = {
+      from,
+      ...transactions[0].params,
+      type: TransactionEnvelopeType.feeMarket,
+    };
+    const securityRequest: ValidateSecurityRequest = {
+      method: 'eth_sendTransaction',
+      params: [trxnParams],
+      origin,
+    };
+    validateSecurity(securityRequest, dappChainId);
```

Comment:
Minor, do we need a `catch` here with a `log` as it's a dangling promise?

Reply:
PPOM validation function `validateSecurity` is handling errors in try/catch block.